### ISSUE TITLE
Managing attachments

### ISSUE THREAD

**Issue by romanofski** [purebred-mua/purebred#10]
Managing attachments during composing as well as reading is important:

Composing:
*On few occasions I think I had to adjust the mime type

Viewing:
* Saving attachments
* Piping attachments through external programs
* Printing individual attachments

**Comment by frasertweedale** [comment]
Saving, yes.  Piping, yes.

IMO printing attachments should be handled by external programs.
I don't think printing as a feature of the MUA itself is important or even
desirable.  (I have never printed from Mutt)

**Comment by romanofski** [comment]
Yep.. I'm with you on everything. Currently printing is done by a2ps.. so maybe it's just a matter of support piping?

**Comment by romanofski** [comment]
I feel like parking this for now, because I don't really want to write more throw away code against the current mime library. Odd thing is anything which is not multipart leads to buggy behavior when it comes to showing the content type: `showType` just shows the entire message.

**Comment by frasertweedale** [comment]
Yeah, we want to add compose and serialisation to purebred-email.  Feel free to have a crack at it.

**Comment by romanofski** [comment]
@frasertweedale would you mind pointing me in the right direction? Maybe just a list of do's and dont's might be enough, unless it's really obvious, then I guess I will be able to figure it out.

**Comment by romanofski** [comment]
Hm... looking through purebred-email, I think most bits are already there. Needs a friendlier API if even that ...

**Comment by frasertweedale** [comment]
Well we don't have serialisation, and we'll definitely need it, so maybe start there, and hopefully
anything else we need you'll discover along the way  :)

**Comment by frasertweedale** [comment]
Agree with this revised scope.

**Comment by romanofski** [comment]
Alright I've got showing entities and piping. The latter needs to correctly decode the entity before writing it to a temporary file and I should have a pull request. For the `saving to disk` item I realised that I want to use the file browser for this, but keep the 'exit action' as a parameter. For that I need to do some refactoring and I think would put the entire item in a separate backlog item. Showing and piping is already valuable in itself.
